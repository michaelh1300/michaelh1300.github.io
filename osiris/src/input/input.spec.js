import { mount } from '@vue/test-utils';
import Input from './input.vue';

describe('Input', () => {
  let wrapper;

  const reset = () => wrapper.setProps({
    name: '',
    placeholder: '',
    width: '300px',
    type: 'text',
    size: 'normal',
    value: '',
    readonly: false,
    autofocus: false,
    disabled: false,
    maxlength: Infinity,
    minlength: 0,
    max: Infinity,
    min: 0,
    prefixIcon: '',
    suffixIcon: '',
    isSuccess: false,
    isError: false,
  });

  beforeAll(() => {
    wrapper = mount(Input);
  });

  afterEach(reset);

  it('should be a default input at startup', () => {
    const defaultClasses = ['o-input', 'o-input--normal'];

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toEqual(defaultClasses);
  });

  it('should be a large input', () => {
    wrapper.setProps({
      size: 'large',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input--large');
  });

  it('should be a small input', () => {
    wrapper.setProps({
      size: 'small',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input--small');
  });

  it('should be a mini input', () => {
    wrapper.setProps({
      size: 'mini',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input--mini');
  });

  it('should be disabled', () => {
    wrapper.setProps({
      disabled: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-disabled');
    expect(wrapper.html()).toContain('disabled=');
  });

  it('should be readonly', () => {
    wrapper.setProps({
      readonly: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.html()).toContain('readonly=');
  });

  it('should have a prefixIcon', () => {
    wrapper.setProps({
      prefixIcon: 'o-icon o-icon-search',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.html()).toContain('o-input__prefix');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-search', 'o-icon--prefix');
  });

  it('should have a suffixIcon', () => {
    wrapper.setProps({
      suffixIcon: 'o-icon o-icon-search',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input__suffix');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-search', 'o-icon--suffix');
  });

  it('should have a success status', () => {
    wrapper.setProps({
      isSuccess: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-success');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-check', 'o-icon--suffix');
  });

  it('should have a error status', () => {
    wrapper.setProps({
      isError: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-error');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-x', 'o-icon--suffix');
  });

  it('should emit a focus event', () => {
    wrapper.findAll('input').trigger('focus');

    expect(wrapper.emitted().focus).toBeTruthy();
  });

  it('should emit a blur event', () => {
    wrapper.findAll('input').trigger('blur');

    expect(wrapper.emitted().blur).toBeTruthy();
  });

  it('should emit a input event', () => {
    wrapper.setProps({
      value: 'abc',
    });
    wrapper.findAll('input').trigger('input');

    expect(wrapper.vm.$data.currentValue).toBe('abc');
    expect(wrapper.emitted().input).toBeTruthy();
  });

  it('should have a suffix icon through slot', () => {
    wrapper = mount(Input, {
      slots: {
        suffix: '<i class="o-icon o-icon-calendar"></i>',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input__suffix');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-calendar', 'o-icon--suffix');
  });

  it('should have a prefix icon through slot', () => {
    wrapper = mount(Input, {
      slots: {
        prefix: '<i class="o-icon o-icon-calendar"></i>',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-input__prefix');
    expect(wrapper.html()).toContain('<i', 'o-icon', 'o-icon-calendar', 'o-icon--prefix');
  });
});
