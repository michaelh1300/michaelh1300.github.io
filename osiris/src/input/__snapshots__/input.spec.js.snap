// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should be a default input at startup 1`] = `
<div class="o-input o-input--normal" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
</div>
`;

exports[`Input should be a large input 1`] = `
<div class="o-input o-input--large" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
</div>
`;

exports[`Input should be a mini input 1`] = `
<div class="o-input o-input--mini" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
</div>
`;

exports[`Input should be a small input 1`] = `
<div class="o-input o-input--small" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
</div>
`;

exports[`Input should be disabled 1`] = `
<div class="o-input o-input--normal is-disabled" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner" disabled="disabled">
  <!---->
  <!---->
</div>
`;

exports[`Input should be readonly 1`] = `
<div class="o-input o-input--normal" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner" readonly="readonly">
  <!---->
  <!---->
</div>
`;

exports[`Input should have a error status 1`] = `
<div class="o-input o-input--normal o-input__suffix is-error" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <i class="o-input__icon--suffix o-icon o-icon-x"></i>
</div>
`;

exports[`Input should have a prefix icon through slot 1`] = `
<div class="o-input o-input--normal o-input__prefix" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
  <i class="o-icon o-icon-calendar"></i>
</div>
`;

exports[`Input should have a prefixIcon 1`] = `
<div class="o-input o-input--normal o-input__prefix" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <i class="o-input__icon--prefix o-icon o-icon-search"></i>
  <!---->
</div>
`;

exports[`Input should have a success status 1`] = `
<div class="o-input o-input--normal o-input__suffix is-success" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <i class="o-input__icon--suffix o-icon o-icon-check"></i>
</div>
`;

exports[`Input should have a suffix icon through slot 1`] = `
<div class="o-input o-input--normal o-input__suffix" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <!---->
  <i class="o-icon o-icon-calendar"></i>
</div>
`;

exports[`Input should have a suffixIcon 1`] = `
<div class="o-input o-input--normal o-input__suffix" style="width: 300px;">
  <input placeholder="" type="text" min="0" max="Infinity" minlength="0" maxlength="Infinity" class="o-input__inner">
  <!---->
  <i class="o-input__icon--suffix o-icon o-icon-search"></i>
</div>
`;
